aspnet/mvc/views/forms/sample/final/Views/Demo/EditColor.cshtml

```diff
@@ -0,0 +1,17 @@
+﻿@model Person
+@{
+    var index = (int)ViewData["index"];
+}
+
+<form asp-controller="ToDo" asp-action="Edit" method="post">
+    @*
+        Move the following two lines to a template to avoid
+        evaluating index twice.
+        <label asp-for="Colors[index]" />
```

Comment:
@Rick-Anderson I don't recommend such a note. We have to assume users know HTML 5 semantics, including the (short) list of "void elements". The closest I'd go would be a note in the Tag Helper document mentioning that no in-box tag helper "fixes" element forms i.e. nothing removes end tags from elements defined to be void or adds end tags for elements defined to be non-void. And I'm not positive about that note either except that @danroth27 and @DamianEdwards probably have opinions.